sources/mma/mma.f90

```diff
@@ -0,0 +1,701 @@
+                    !the residue
+                    rex(:) = ((this%p0j(:) + matmul(transpose(this%pij(:,:)), &
+                            lambda(:)))/(this%upp(:) - x(:))**2 - &
+                            (this%q0j(:) + matmul(transpose(this%qij(:,:)), &
+                            lambda(:)))/(x(:) - this%low(:))**2 ) - &
+                            xsi(:) + eta(:)
+                    rey(:) = this%c(:) + this%d(:)*y(:) - lambda(:) - mu(:)
+                    rez = this%a0 - zeta - dot_product(lambda(:) , this%a(:))
+                    relambda(:) = matmul(this%pij(:,:),1.0/&
+                            (this%upp(:)-x(:))) + matmul(this%qij(:,:), &
+                            1.0/(x(:)-this%low(:))) - this%a(:)*z - &
+                            y(:) + s(:) - this%bi(:)
+                    rexsi(:) = xsi(:)*(x(:)-this%alpha(:)) - epsi
+                    reeta(:) = eta(:)*(this%beta(:)-x(:)) - epsi
+                    remu(:) = mu(:)*y(:) - epsi
+                    rezeta = zeta*z - epsi
+                    res(:) = lambda(:)*s(:) - epsi
+                    
+                    residu = [rex, rey, rez, relambda, &
+                                rexsi, reeta, remu, rezeta, res]
+                    newresidu = norm2(residu)
+                    !print *, residunorm
+                    !print *, newresidu
+                    steg = steg/2
+                end do
+
+                residunorm = newresidu
+                residumax = maxval(abs(residu))
+                !correct the step size for the extra devision by 2 in the final
+                !loop
+                steg = 2*steg
+                !print *, "steg=", steg, "residumax=", residumax, "epsi=", epsi
+            end do
+            epsi = 0.1*epsi
+        end do
+        this%x=x
+
+        !update the parameters of the MMA object nesessary to compute KKT residu
+        this%y=y
+        this%z=z
+        this%lambda=lambda
+        this%zeta=zeta
+        this%xsi=xsi
+        this%eta=eta
+        this%mu=mu
+        this%s=s
+
+
+        ! print *, "x=", x
+        ! print *, "y=", y
+        ! print *, "z=", z
+        ! print *, "lambda=", lambda
+        ! print *, "xsi=", xsi
+        ! print *, "eta=", eta
+        ! print *, "mu=", mu
+        ! print *, "zeta=", zeta
+        ! print *, "s=", s
+
+        !print *, steg
+
+        !print *, dy
+        !print *, dxsi
+        !print *, deta
+        !print *, dmu
+        !print *, dzeta
+        !print *, ds
+        !print *, info
+        !print *, steg
+
+        ! print *, delx
+        ! print *, dely
+        ! print *, delz
+        ! print *, dellambda
+        !print *, relambda
+        !print *, residumax
+        !print *, norm2(residu)
+        !print *, residu
+
+        ! print *, this%alpha
+        ! print *, x
+        ! print *, this%beta
+        ! print *, xsi(:)*(x(:)-this%alpha(:))
+        ! print *, epsi
```

Comment:
Please remove unused comments.
Or add it as prints under a debug flag if you need to keep it.